src/with-theme.js

```diff
@@ -0,0 +1,57 @@
+import React, {Component} from 'react'
+
+import {CHANNEL} from './theme-provider'
+import {PropTypes} from './react-compat'
+
+function generateWarningMessage(componentName) {
+  // eslint-disable-next-line max-len
+  return `glamorous warning: Expected component called ${componentName} which uses withTheme to be within a ThemeProvider but none was found.`
+}
+
+export default function withTheme(ComponentToTheme) {
+  class ThemedComponent extends Component {
+    state = {theme: {}}
+    setTheme = theme => this.setState({theme})
+
+    componentWillMount() {
+      if (!this.context[CHANNEL]) {
+        // istanbul ignore else
```

Comment:
Actually, I'd really like to make sure that we don't log in production. So let's add a test that basically does:

```javascript
const originalEnv = process.env.NODE_ENV
process.env.NODE_ENV = 'production'

// tests

process.env.NODE_ENV = originalEnv
```

Thanks!